include/zephyr/arch/riscv/syscall.h

```diff
@@ -113,35 +130,47 @@ static inline uintptr_t arch_syscall_invoke2(uintptr_t arg1, uintptr_t arg2,
 {
 	register ulong_t a0 __asm__ ("a0") = arg1;
 	register ulong_t a1 __asm__ ("a1") = arg2;
-	register ulong_t a7 __asm__ ("a7") = call_id;
+#if defined(CONFIG_RISCV_ISA_RV32E)
+	register ulong_t syscall_id __asm__ ("t0") = call_id;
+#else
+	register ulong_t syscall_id __asm__ ("a7") = call_id;
+#endif /* CONFIG_RISCV_ISA_RV32E */
 
 	__asm__ volatile ("ecall"
 			  : "+r" (a0)
-			  : "r" (a1), "r" (a7)
+			  : "r" (a1), "r" (syscall_id)
 			  : "memory");
 	return a0;
 }
 
 static inline uintptr_t arch_syscall_invoke1(uintptr_t arg1, uintptr_t call_id)
 {
 	register ulong_t a0 __asm__ ("a0") = arg1;
-	register ulong_t a7 __asm__ ("a7") = call_id;
+#if defined(CONFIG_RISCV_ISA_RV32E)
+	register ulong_t syscall_id __asm__ ("t0") = call_id;
+#else
+	register ulong_t syscall_id __asm__ ("a7") = call_id;
+#endif /* CONFIG_RISCV_ISA_RV32E */
```

Comment:
Nitpick: this `if defined` repeats a lot here, perhaps hide it behind a macro `SAVE_CALL_ID` or so?

Reply:
yup, makes sense.